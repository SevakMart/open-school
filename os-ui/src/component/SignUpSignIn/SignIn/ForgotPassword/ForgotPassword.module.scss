@import "../../../../styles/Mixins/ButtonMixin.scss";
@import "../../../../styles/Mixins/FlexMixins.scss";
@import "../../../../styles/Mixins/FontMixin.scss";

.mainContainer {
    @include flexStyle(column, flex-start, space-between);
    margin-top: 3%;
    margin-bottom: 1%;
    padding-left: 8%;
    padding-right: 8%;
    padding-top: 4%;
    padding-bottom: 1%;

    h2{
        @include fontStyle("Montserrat", normal, 800, 2.25rem, 2.75rem);
        color: #222427;
        align-self:center;
    }

    div {
      position: relative;
      width: 100%;
      margin-top: 5%;
      @include flexStyle(column, flex-start, space-between);
    }
    input {
      width: 100%;
      border: 1px solid #d9dbe9;
      box-sizing: border-box;
      border-radius: 0.75rem;
      height: 3.5rem;
      margin-top: 2%;
      margin-bottom: 3%;
      padding-left: 3%;
    }
    label {
      @include fontStyle("Montserrat", normal, 500, 0.875rem, 1rem);
      color: #333941;
    }
    p {
      display: block;
      width: 100%;
      text-align: center;
      @include fontStyle("Montserrat", normal, 500, 1rem, 0.0rem);
      color: #333941;
      cursor: pointer;
    }
    button {
      @include buttonStyle(#5e617b,#ffffff,1rem,100%,3.5rem,none,none,none,none);
      @include fontStyle("Montserrat", normal, 500, 1rem, 1.25rem);
      display: block;
      text-align: center;
      border: 1px solid #5e617b;
      box-sizing: border-box;
      margin-top: 4%;
    }
  }

  .successedMessage {
    @include fontStyle("Montserrat", normal, 700, 1.25rem, 2.5rem);
    margin-top: 5%;
    margin-left: 5%;
    margin-right: 5%;
    color: #222427;
    text-align: center
  }

  .errorMessage {
    @include fontStyle("Montserrat", normal, 800, 0.875rem, 0rem);
    color: #ff0000;
  }

