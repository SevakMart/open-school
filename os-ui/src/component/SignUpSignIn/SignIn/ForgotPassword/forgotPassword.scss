@import './../../../../styles/Mixins/ButtonMixin.scss';
@import './../../../../styles/Mixins/FontMixin.scss';
@import './../../../../styles/Configs/Fonts.scss';
@import './../../../../styles/Configs/Colors.scss';

.closeIconX {
  position: absolute;
  top: 24px;
  right: 24px;
  box-sizing: border-box;
  width: 24px;
  height: 24px;
  cursor: pointer;
  &:hover {
    color: brown;
  }  
}

.forgotpsd_box {
  font-family: $primary-font;
  margin-top: 35px;
  &_title {
    text-align: center;
    font-weight: 800;
    font-size: 2.25rem;
    color:$header-main-color;
  }
  &_subtitles {
   margin-top: 20px; 
  }
  &_subtitle {
    font-size: 1rem;
    color:$main-text-color;
    margin-top: 10px;
  }
}

.ForgotPasswordBtns {
  display: flex;
  justify-content: flex-start;
  gap: 20px;
  margin-top: 40px;
}

.anyBtn {
  padding: 20px 0;
  width: 100%;
  font-family: $primary-font;
  font-weight: 500;
  font-size: 18px;
  border-radius: 0.7rem;
  background-color: $secondary-color;
  color: $white-text-color;
  display: block;
  text-align: center;
  box-sizing: border-box;
  border: none;
  margin-bottom: 24px;
}

.backButton {
  order: 1;
  width: 100%;
  margin-right: auto;
}

.continueButton {
  order: 2;
}

.anyBtn,
.backButton {
  &:hover {
    background-color: #83f0dc;
  }
  &:active {
    background-color: #17ac90;
  }
  &:hover[disabled] {
    background-color: $btn-primary-color;
    color: #fff;
  }

  &[disabled] {
    opacity: 0.5;
    cursor: not-allowed;
  }
}

.forgotpsdContainer {
  margin: 20px 80px 0 80px;
  &_mainBody {
    margin-top: 30px;    
  }
}

@media screen and (max-width: 1080px) {
  .anyBtn {
    font-size: 14px;
    padding: 15px 10px;
  }
}

@media screen and (max-width: 900px) {
  .ForgotPasswordBtns {
    display: block;
  }
}