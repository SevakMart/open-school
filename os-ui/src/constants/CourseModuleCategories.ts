export const COURSE_CATEGORIES = [
  {
    value: '1',
    id: '1',
    title: 'Module 1',
  },
  {
    value: '2',
    id: '2',
    title: 'Module 2',
  },
  {
    value: '3',
    id: '3',
    title: 'Module 3',
  },
  {
    value: '4',
    id: '4',
    title: 'Module 4',
  },
];

export const COURSE_NAME = [
  {
    id: '1',
    desc: 'Additional Materials',
  },
  {
    id: '2',
    desc: 'Grade',
  },
  {
    id: '3',
    desc: 'Discussion Form',
  },
];
