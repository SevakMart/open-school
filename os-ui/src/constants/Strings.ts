// Navbar strings
export const APP_LOGO = 'Open-School';
export const EXPLORE_CATEGORIES = 'Explore Categories';
export const MENTORS = 'Mentors';
export const BECOME_A_MENTOR = 'Become a mentor';
export const SIGN_IN = 'Sign In';
export const ALL_LEARNING_PATHS = 'All Learning Paths';
export const MY_LEARNING_PATHS = 'My Learning Paths';
export const ALL = 'All';
export const IN_PROGRESS = 'InProgress';
export const COMPLETED = 'Completed';

// Header strings
export const FREE_EDUCATIONAL_PLATFORM = 'Educational Platform';
export const SIGN_UP = 'SIGN UP';
export const HEADER_INTRODUCTION = 'Choose Categories You Are Interested In Choose Categories You Are Interested InChoose Categories You Are Interested';
export const EDUCATION_PLATFORM_IMAGE = 'https://image.shutterstock.com/image-vector/ebook-logo-design-vector-electronic-260nw-1767373559.jpg';
export const CHOOSE_CATEGORIES_HEADER = 'Choose Categories You Are Interested In';

// URLS
export const GET_MENTORS_URL = '../../mentorData.json';
export const GET_CATEGORIES_URL = '../../categorieData.json';
export const GET_REAL_MENTORS_URL = `http://${process.env.REACT_APP_HOSTNAME}/api/v1/users/mentors?`;
export const REGISTRATION_URL = `http://${process.env.REACT_APP_HOSTNAME}/api/v1/auth/register`;
export const GET_MAIN_CATEGORIES_URL = `http://${process.env.REACT_APP_HOSTNAME}/api/v1/categories?`;
export const GET_CATEGORY_SUBCATEGORY_SEARCH_URL = `http://${process.env.REACT_APP_HOSTNAME}/api/v1/categories/subcategories?title=`;
export const SIGNIN_URL = `http://${process.env.REACT_APP_HOSTNAME}/api/v1/auth/login`;
export const SAVE_PREFERRED_CATEGORIES = `http://${process.env.REACT_APP_HOSTNAME}/api/v1/users`;
export const USER_URL = `http://${process.env.REACT_APP_HOSTNAME}/api/v1/users`;
// Error Message
export const ERROR_MESSAGE = 'Something went wrong please refresh the page :(';
export const EMPTY_DATA_ERROR_MESSAGE = 'No data to display :(';

// Successful Sign in message
export const SUCCESSFUL_SIGNIN_MESSAGE = 'You have Successfully signed in!';

// Regex for form validation
export const fullNameRegex = /^[A-Z\s]*$/i;
export const emailRegex = /^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$/g;
export const passwordRegex = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8,}$/;
/* Userâ€™s Password should have at least 8 characters,
including upper and lower case letters, numbers and signs (required) */

// Form Validation error message
export const INVALID_FULL_NAME_ERROR_MESSAGE = 'Invalid name or surname';
export const INVALID_EMAIL_ERROR_MESSAGE = 'Invalid email address';
export const INVALID_PASSWORD_ERROR_MESSAGE = 'Password should be at least 8 characters and contain at least one uppercase,lowercase,number and special character';
export const FULL_NAME_TOO_SHORT = 'full name is too short';
export const EMAIL_TOO_SHORT = 'email is too short';
export const PASSWORD_TOO_SHORT = 'password is too short';
export const PASSWORD_REQUIRED = 'password is required';
export const EMAIL_REQUIRED = 'email is required';

// Other non-specific strings
export const NO_COURSES_YET = 'No Courses Yet';
export const EXPLORE_COURSES = 'Explore Courses';
export const SUGGESTED_LEARNING_PATHS = 'Suggested Learning Paths';
export const OVERALL_PROGRESS = 'Overall Progress';
export const REMAINING_TIME = 'Remaining Time';
export const STATUS = 'Status';
export const DUE_DATE = 'Due Date';
export const RESUME_COURSE = 'Resume Course';
export const GRADE = 'Grade';
export const RATE_COURSE = 'Rate Course';
