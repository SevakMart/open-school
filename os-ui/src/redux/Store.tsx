import { configureStore } from '@reduxjs/toolkit';
import {
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import { createStateSyncMiddleware, initMessageListener } from 'redux-state-sync';
import rootReducer from './RootReducer';

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: {
      ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
    },
  }).concat(
    createStateSyncMiddleware({
      blacklist: ['persist/PERSIST', 'persist/REHYDRATE'],
    }),
  ),
});

initMessageListener(store);

export type RootState = ReturnType<typeof store.getState>
