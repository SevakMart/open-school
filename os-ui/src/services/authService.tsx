import fetchService from './fetchData';
import { RegistrationFormType } from '../types/RegistartionFormType';
import { ResetPasswordType } from '../types/ResetPasswordType';

class AuthService {
  readonly basePath: string;

  constructor() {
    this.basePath = 'auth';
  }

  async register(content:RegistrationFormType) {
    const response = await fetchService.post(`${this.basePath}/register`, content, {});
    const { status } = response;
    const data = await response.json();
    return { ...data, status };
  }

  async signIn(content:Omit<RegistrationFormType, 'firstName' | 'lastName'>) {
    const response = await fetchService.post(`${this.basePath}/login`, content, {});
    const { status, headers } = response;
    const data = await response.json();
    const FullTokenResponse = headers.get('Authorization');
    const tokenStartingIndex = FullTokenResponse ? FullTokenResponse.indexOf(' ') : undefined;
    const token = FullTokenResponse && tokenStartingIndex
      ? FullTokenResponse.slice(tokenStartingIndex + 1) : null;
    return { ...data, status, token };
  }

  async redirectUserToAccount(content:object) {
    const response = await fetchService.post(`${this.basePath}/account`, content, {});
    const { status, headers } = response;
    let data = {};
    if (response.status === 200) {
      data = await response.json();
    }
    const FullTokenResponse = headers.get('Authorization');
    const tokenStartingIndex = FullTokenResponse ? FullTokenResponse.indexOf(' ') : undefined;
    const token = FullTokenResponse && tokenStartingIndex
      ? FullTokenResponse.slice(tokenStartingIndex + 1) : null;
    return { ...data, status, token };
  }

  async sendForgotPasswordRequest(content:object) {
    const response = await fetchService.post(`${this.basePath}/password/forgot`, content, {});
    const { status } = response;
    const data = await response.json();
    return { data, status };
  }

  async sendResetPasswordRequest(content:ResetPasswordType) {
    const response = await fetchService.post(`${this.basePath}/password/reset`, content, {});
    const { status } = response;
    const data = await response.json();
    return { data, status };
  }
}

export default new AuthService();
