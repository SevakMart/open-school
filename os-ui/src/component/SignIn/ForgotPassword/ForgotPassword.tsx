import React, { useEffect, useState } from 'react';
import { validateEmail } from '../../../helpers/EmailValidate';
import authService from '../../../services/authService';
import ResetPassword from '../ResetPassword/ResetPassword';
import {
  EMAIL, FORGOT_PASSWORD, ENTER_EMAIL_FOR_VERIFICATION, SEND_CODE_NOTIFICATION,
  CONTINUE,
} from '../../../constants/Strings';
import CloseIcon from '../../../icons/Close';
import styles from './ForgotPassword.module.scss';

const ForgotPassword = (
  { returnToSignInForm }: {returnToSignInForm: ()=> void},
) => {
  const [email, setEmail] = useState('');
  const [successMessage, setSuccessMessage] = useState('');
  const [emailError, setEmailError] = useState('');
  const [resetPassword, setResetPassword] = useState(false);
  const { mainContainer, errorMessage, successedMessage } = styles;

  const handleEmailChange = (e:React.SyntheticEvent) => {
    setEmail((e.target as HTMLInputElement).value);
  };

  const sendForgotPassword = () => {
    const emailError = (validateEmail(email));
    if (!emailError) {
      authService.sendForgotPasswordRequest({ email })
        .then((response) => {
          if (response.status === 200) {
            setSuccessMessage(response.data.message);
            setEmailError('');
          } else if (response.status === 400) {
            setEmailError(response.data.message);
          }
        });
    } else setEmailError(emailError);
  };
  useEffect(() => {
    let timer:any;
    if (successMessage) {
      timer = setTimeout(() => setResetPassword(true), 3500);
    }
    return () => clearTimeout(timer);
  }, [successMessage]);

  return (
    <>
      {
        resetPassword ? (
          <ResetPassword
            returnToSignInForm={() => returnToSignInForm()}
            email={email}
          />
        )
          : (
            <>
              {
              successMessage ? <h3 data-testid="forgotPasswordSuccessMessage" className={successedMessage}>{successMessage}</h3>
                : (
                  <>
                    <CloseIcon handleClosing={() => returnToSignInForm()} />
                    <div className={mainContainer}>
                      <h2>{FORGOT_PASSWORD}</h2>
                      <p>{ENTER_EMAIL_FOR_VERIFICATION}</p>
                      <p>{SEND_CODE_NOTIFICATION}</p>
                      <div>
                        <label htmlFor="email">
                          {EMAIL}
                          <span style={{ color: 'red' }}> *</span>
                        </label>
                        <input
                          data-testid="forgotPasswordEmailInput"
                          id="email"
                          type="email"
                          name="email"
                          placeholder="ex: namesurname@gmail.com"
                          onChange={handleEmailChange}
                          required
                        />
                      </div>
                      {emailError ? (<h4 data-testid="emailErrorForgotPassword" className={errorMessage}>{emailError}</h4>) : null}
                      <button data-testid="continueButton" type="button" onClick={sendForgotPassword}>{CONTINUE}</button>
                    </div>
                  </>
                )
                }
            </>
          )
}
    </>
  );
};
export default ForgotPassword;
