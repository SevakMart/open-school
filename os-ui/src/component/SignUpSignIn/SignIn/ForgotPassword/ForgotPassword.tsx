import React, { useEffect, useState } from 'react';
import { useTranslation } from 'react-i18next';
import { validateEmail } from '../../../../helpers/EmailValidate';
import authService from '../../../../services/authService';
import ResetPassword from '../ResetPassword/ResetPassword';
import CloseIcon from '../../../../icons/Close';
import styles from './ForgotPassword.module.scss';

const ForgotPassword = (
  { returnToSignInForm }: {returnToSignInForm: ()=> void},
) => {
  const { t } = useTranslation();
  const [email, setEmail] = useState('');
  const [successMessage, setSuccessMessage] = useState('');
  const [emailError, setEmailError] = useState('');
  const [resetPassword, setResetPassword] = useState(false);
  const { mainContainer, errorMessage, successedMessage } = styles;

  const handleEmailChange = (e:React.SyntheticEvent) => {
    setEmail((e.target as HTMLInputElement).value);
  };

  const sendForgotPassword = () => {
    const emailError = (validateEmail(email));
    if (!emailError) {
      authService.sendForgotPasswordRequest({ email })
        .then((response) => {
          if (response.status === 200) {
            setSuccessMessage(response.data.message);
            setEmailError('');
          } else if (response.status === 400) {
            setEmailError(response.data.message);
          }
        });
    } else setEmailError(emailError);
  };
  useEffect(() => {
    let timer:any;
    if (successMessage) {
      timer = setTimeout(() => setResetPassword(true), 3500);
    }
    return () => clearTimeout(timer);
  }, [successMessage]);

  return (
    <>
      {
        resetPassword ? (
          <ResetPassword
            returnToSignInForm={() => returnToSignInForm()}
            email={email}
          />
        )
          : (
            <>
              {
              successMessage ? <h3 data-testid="forgotPasswordSuccessMessage" className={successedMessage}>{successMessage}</h3>
                : (
                  <>
                    <CloseIcon />
                    <div className={mainContainer}>
                      <h2>{t('string.forgotPsd.title')}</h2>
                      <p>{t('string.forgotPsd.enterEmail')}</p>
                      <p>{t('string.forgotPsd.sendNotification')}</p>
                      <div>
                        <label htmlFor="email">
                          {t('form.labels.email')}
                          <span style={{ color: 'red' }}> *</span>
                        </label>
                        <input
                          data-testid="forgotPasswordEmailInput"
                          id="email"
                          type="email"
                          name="email"
                          placeholder={t('form.placeholder.email')}
                          onChange={handleEmailChange}
                          required
                        />
                      </div>
                      {emailError ? (<h4 data-testid="emailErrorForgotPassword" className={errorMessage}>{emailError}</h4>) : null}
                      <button data-testid="continueButton" type="button" onClick={sendForgotPassword}>{t('button.forgotPsd.continue')}</button>
                    </div>
                  </>
                )
                }
            </>
          )
}
    </>
  );
};
export default ForgotPassword;
