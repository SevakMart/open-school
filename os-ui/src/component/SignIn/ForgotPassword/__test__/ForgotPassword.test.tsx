import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import { Provider } from 'react-redux';
import { store } from '../../../../redux/Store';
import { EMAIL_REQUIRED, INVALID_EMAIL_ERROR_MESSAGE } from '../../../../constants/Strings';
import ForgotPassword from '../ForgotPassword';
import authService from '../../../../services/authService';

const successData = {
  data: {
    message: 'Email was send successfully',
  },
  status: 200,
};
const failedData = {
  data: {
    message: 'This email does not exist in the database',
  },
  status: 400,
};

describe('Create test cases for ForgotPassword component', () => {
  test('Create a snapshot test', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <ForgotPassword returnToSignInForm={() => null} />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  test('Create a test where we click continue without adding an email on input field', () => {
    expect.hasAssertions();
    render(
      <Provider store={store}>
        <ForgotPassword returnToSignInForm={() => null} />
      </Provider>,
    );
    const continueButton = screen.queryByTestId('continueButton');

    userEvent.click((continueButton as HTMLButtonElement));

    const emailErrorForgotPasswordElement = screen.queryByTestId('emailErrorForgotPassword');

    expect(emailErrorForgotPasswordElement).toBeInTheDocument();
    expect(emailErrorForgotPasswordElement).toHaveTextContent(EMAIL_REQUIRED);
  });
  test('Input an invalid email address', () => {
    expect.hasAssertions();
    render(
      <Provider store={store}>
        <ForgotPassword returnToSignInForm={() => null} />
      </Provider>,
    );
    const forgotPasswordEmailInputElement = screen.queryByTestId('forgotPasswordEmailInput');
    const continueButton = screen.queryByTestId('continueButton');

    userEvent.type((forgotPasswordEmailInputElement as HTMLInputElement), 'john');
    userEvent.click((continueButton as HTMLButtonElement));

    const emailErrorForgotPasswordElement = screen.queryByTestId('emailErrorForgotPassword');

    expect(emailErrorForgotPasswordElement).toBeInTheDocument();
    expect(emailErrorForgotPasswordElement).toHaveTextContent(INVALID_EMAIL_ERROR_MESSAGE);
  });
  test('Create a successful mail sending', async () => {
    expect.hasAssertions();
    jest.spyOn(authService, 'sendForgotPasswordRequest').mockResolvedValue(successData);

    render(
      <Provider store={store}>
        <ForgotPassword returnToSignInForm={() => null} />
      </Provider>,
    );
    const forgotPasswordEmailInputElement = screen.queryByTestId('forgotPasswordEmailInput');
    const continueButton = screen.queryByTestId('continueButton');

    userEvent.type((forgotPasswordEmailInputElement as HTMLInputElement), 'john@john.com');
    userEvent.click((continueButton as HTMLButtonElement));

    const forgotPasswordSuccessMessageElement = await screen.findByTestId('forgotPasswordSuccessMessage');

    expect(forgotPasswordSuccessMessageElement).toBeInTheDocument();
    expect(forgotPasswordSuccessMessageElement).toHaveTextContent(successData.data.message);
  });

  test('Create a successful mail sending', async () => {
    expect.hasAssertions();
    jest.spyOn(authService, 'sendForgotPasswordRequest').mockResolvedValue(failedData);

    render(
      <Provider store={store}>
        <ForgotPassword returnToSignInForm={() => null} />
      </Provider>,
    );
    const forgotPasswordEmailInputElement = screen.queryByTestId('forgotPasswordEmailInput');
    const continueButton = screen.queryByTestId('continueButton');

    userEvent.type((forgotPasswordEmailInputElement as HTMLInputElement), 'john@john.com');
    userEvent.click((continueButton as HTMLButtonElement));

    const emailErrorForgotPasswordElement = await screen.findByTestId('emailErrorForgotPassword');

    expect(emailErrorForgotPasswordElement).toBeInTheDocument();
    expect(emailErrorForgotPasswordElement).toHaveTextContent(failedData.data.message);
  });
});
