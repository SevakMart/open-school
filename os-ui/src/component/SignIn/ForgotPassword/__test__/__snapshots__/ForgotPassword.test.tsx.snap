// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create test cases for ForgotPassword component Create a snapshot test 1`] = `
<DocumentFragment>
  <div
    data-testid="closeButton"
    style="background-color: transparent; color: rgb(132, 138, 157); cursor: pointer; float: inline-end; margin-right: 3%; font-size: 2rem; padding: 0px;"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7,7 L17,17 M7,17 L17,7"
        fill="none"
        stroke="#000"
        stroke-width="2"
      />
    </svg>
  </div>
  <div
    class="mainContainer"
  >
    <h2>
      Forgot password
    </h2>
    <p>
      Enter your email for the verification process.
    </p>
    <p>
      We will send 4 digits code to your email.
    </p>
    <div>
      <label
        for="email"
      >
        Email
        <span
          style="color: red;"
        >
           *
        </span>
      </label>
      <input
        data-testid="forgotPasswordEmailInput"
        id="email"
        name="email"
        placeholder="ex: namesurname@gmail.com"
        required=""
        type="email"
      />
    </div>
    <button
      data-testid="continueButton"
      type="button"
    >
      Continue
    </button>
  </div>
</DocumentFragment>
`;
